src/gui/src/qpxgraph.cpp

```diff
@@ -0,0 +1,1403 @@
+/*
+ * This file is part of the QPxTool project.
+ * Copyright (C) 2008-2010 Gennady "ShultZ" Kozlov <qpxtool@mail.ru>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * See the file "COPYING" for the exact licensing terms.
+ *
+ */
+
+#include <QContextMenuEvent>
+#include <QMenu>
+#include <QPaintEvent>
+#include <QPainter>
+#include <QPainterPath>
+
+#include "qpxgraph.hpp"
+#include <cmath>
+
+#include <QDebug>
+
+#include <sys/time.h>
+
+#define SHOW_P95ERRC
+//#define SHOW_P95JB
+
+#define MARGIN_DEFL 40
+#define MARGIN_DEFR 40
+#define MARGIN_DEFB 18
+
+//#define GRID_STYLE Qt::DotLine
+#define GRID_STYLE Qt::DashLine
+
+class IntList : public QList<int> {
+public:
+  IntList() : QList<int>(){};
+
+  float M() {
+    float M = 0.0;
+    int s = size();
+    if (!s)
+      return 0.0;
+    for (int i = 0; i < s; i++)
+      M += at(i);
+    return (float)(M / s);
+  };
+  float dispers(float M) {
+    double disp = 0.0;
+    double q;
```

Comment:
Codacy found an issue: [The scope of the variable 'q' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)